core/dbt/task/show.py

```diff
@@ -23,6 +23,11 @@ def execute(self, compiled_node, manifest):
         # Allow passing in -1 (or any negative number) to get all rows
         limit = None if self.config.args.limit < 0 else self.config.args.limit
 
+        if "sql_header" in compiled_node.unrendered_config:
+            compiled_node.compiled_code = (
+                compiled_node.unrendered_config["sql_header"] + compiled_node.compiled_code
```

Comment:
Almost certainly a premature optimization, but it's worth pointing out that string concatenation with the `+` operator is a bit inefficient.